helm/cloudflared/templates/pre-delete-job.yaml

```diff
@@ -0,0 +1,41 @@
+{{- if not .Values.useExistingTunnels.enabled }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "cloudflared-{{ .Release.Name }}-helm-pre-delete"
+  labels:
+    app.kubernetes.io/name: "cloudflared-{{ .Release.Name }}"
+    {{- include "labels.common" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-weight": "-1"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: "cloudflared-{{ .Release.Name }}"
+        {{- include "labels.selector" . | nindent 8 }}
+    spec:
+      serviceAccountName: "{{ .Chart.Name }}-{{ .Release.Name }}"
+      restartPolicy: Never
+      containers:
+      - name: clean-delete-tunnel
+        image: {{ .Values.initImage.registry }}/{{ .Values.initImage.name }}:{{ .Values.initImage.tag }}
+        imagePullPolicy: {{ .Values.initImage.pullPolicy }}
+        securityContext:
+          runAsUser: 1000
+          readOnlyRootFilesystem: true
```

Comment:
IMO this will also make `/tmp` read-only.

Reply:
Good catch @ubergesundheit, thanks for pointing it out. I've added commit 5331bb9 that mounts an emptydir for the scripts.